packages/template-retail-react-app/app/commerce-api/hooks/useBasket.js

```diff
@@ -115,9 +105,20 @@ export default function useBasket(opts = {}) {
              * @param {number} item.quantity - The quantity of the item.
              */
             async addItemToBasket(item) {
+                let newBasket = {}
+                if (!basket || !basket.basketId) {
+                    // Back to using ShopperBaskets for all basket interaction.
+                    newBasket = await api.shopperBaskets.createBasket({})
+
+                    // Throw if there was a problem creating the basket
+                    if (isError(basket)) {
+                        throw new Error(basket)
+                    }
+                }
+
                 const response = await api.shopperBaskets.addItemToBasket({
                     body: item,
-                    parameters: {basketId: basket.basketId}
+                    parameters: {basketId: basket?.basketId || newBasket.basketId}
```

Comment:
Prioritize newBasket since we know it will be there, once it is not, we can be certain the basket should be assumed to be valid
```suggestion
                    parameters: {newBasket.basketId || basketId: basket.basketId}
```

Reply:
Same as above.
Reverted useBasket back to original code.